src/formatDate.js

```diff
@@ -50,7 +50,65 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  let day;
+  let month;
+  let unfixedYear;
+  let newFormatYear;
+
+  for (let i = 0; i < 3; i++) {
+    switch (fromFormat[i]) {
+      case 'DD':
+        day = dateParts[i];
+        break;
+
+      case 'MM':
+        month = dateParts[i];
+        break;
+
+      case 'YY':
+      case 'YYYY':
+        unfixedYear = dateParts[i];
+        break;
+    }
```

Comment:
You do the same operation in all cases. Let's create an array `newDateFormat` and push into it some date parts, without this `switch cases `

Reply:
I have no ideas how to reorder dates without switch cases :(